modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -367,45 +367,99 @@ void ONNXImporter::populateNet(Net dstNet)
+                DictValue ends = layerParams.get("ends");
+                CV_Assert(starts.size() == ends.size());
+
+                if (axis > 0) {
+                    begin.resize(axis, 0);
+                    end.resize(axis, -1);
+                }
+                for (int i = 0; i < starts.size(); ++i)
+                {
+                    begin.push_back(starts.get<int>(i));
+                    int finish = ends.get<int>(i);
+                    end.push_back((finish < 0) ? --finish : finish); // numpy doesn't include last dim
+                }
+            } else {
+                CV_Assert(inp_size >= 3);
+                for (int i = 1; i < inp_size; i++) {
+                    CV_Assert(constBlobs.find(node_proto.input(i)) != constBlobs.end());
+                }
+                Mat start_blob = getBlob(node_proto, constBlobs, 1);
+                Mat end_blob   = getBlob(node_proto, constBlobs, 2);
+                CV_Assert(start_blob.total() == end_blob.total());
+
+                if (inp_size > 3) {
+                    Mat axes_blob = getBlob(node_proto, constBlobs, 3);
+                    const int* axes = (int*)axes_blob.data;
+                    for (int i = 1; i < axes_blob.total(); ++i) {
+                        CV_Assert(axes[i - 1] == axes[i] - 1);
+                    }
+                    axis = axes[0];
                 }
-                axis = axes.get<int>(0);
-            }
-            layerParams.set("axis", axis);
 
-            DictValue starts = layerParams.get("starts");
-            DictValue ends = layerParams.get("ends");
-            CV_Assert(starts.size() == ends.size());
+                const int* starts = (int*)start_blob.data;
+                const int* ends = (int*)end_blob.data;
+                if (axis > 0) {
+                    begin.resize(axis, 0);
+                    end.resize(axis, -1);
+                }
+                std::copy(starts, starts + start_blob.total(), std::back_inserter(begin));
+                for (int i = 0; i < end_blob.total(); ++i)
+                {
+                    int finish = ends[i];
+                    end.push_back((finish < 0) ? --finish : finish); // numpy doesn't include last dim
+                }
 
-            std::vector<int> begin;
-            std::vector<int> end;
-            if (axis > 0) {
-                begin.resize(axis, 0);
-                end.resize(axis, -1);
+                if (inp_size == 5) {
+                    CV_Assert(constBlobs.find(node_proto.input(4)) != constBlobs.end());
+                    Mat step_blob = getBlob(node_proto, constBlobs, 4);
+                    double min_val, max_val;
+                    minMaxLoc(getBlob(node_proto, constBlobs, 4), &min_val, &max_val, NULL, NULL);
+                    if (min_val != max_val || min_val != 1) {
+                        CV_Error(Error::StsNotImplemented,
+                            "Slice layer only supports steps = 1");
+                    }
```

Comment:
Maybe something like
```cpp
CV_CheckEQ(countNonZero(step_blob != 1), 0, "Slice layer only supports steps = 1");
```

Reply:
fixed

Reply:
But this way we do not need to compute max and mix values.